### PR TITLE
Whitelist user access to reader

### PR DESCRIPTION
**Pull request by joofsh**
This PR adds support for whitelisting users to Reader. 

connects https://github.com/department-of-veterans-affairs/caseflow/issues/1831

TODO:
- [x] Update Gemfile with new caseflow-commons version

CC @mdbenjam for reader pilot awareness

### Plans for deploying this:
- Before deploy, go into prod console and globally enable reader `FeatureToggle.enable!(:reader)`
- After deploy, go into prod console and whitelist specific users that we know should have access
`FeatureToggle.enable!(:reader, users: [...])`

### Test Plan
- [x] Locally enabled the feature globally, verified a user could access reader
- [x] Locally enabled feature but only for specific users, verified they had access but other users don't
- [x] Added tests

### CODE REVIEW

File: lib/fakes/appeal_repository.rb [hunk 1]
```diff
@@ -323,6 +323,8 @@ def self.random_reader_documents(num_documents)
   end
 
   def self.seed_reader_data!
+    FeatureToggle.enable!(:reader)
```
Comment (joofsh) [inline lib/fakes/appeal_repository.rb]: this keeps it from disrupting people in dev/demo mode

Comment (amprokop) [inline lib/fakes/appeal_repository.rb]: cool

File: spec/feature/reader_spec.rb [hunk 2]
```diff
@@ -15,8 +15,14 @@ def in_viewport(element)
 end
 
 RSpec.feature "Reader" do
+  before do
+    FeatureToggle.disable!(:reader)
+    FeatureToggle.enable!(:reader)
```
Comment (amprokop) [inline spec/feature/reader_spec.rb]: why is this line necessary?

Comment (amprokop) [inline spec/feature/reader_spec.rb]: it seems like either `FeatureToggle.disable!(:reader)` or `FeatureToggle.enable!(:reader)` could be what we want, not sure why we'd need both. mind explaining?

Comment (joofsh) [inline spec/feature/reader_spec.rb]: So if a test does `FeatureToggle.enable!(:reader, users: [1,2,3])`, then a subsequent call to `FeatureToggle.enable!(:reader)` is a no-op, because it would be "removing" the users restriction that previously existed. If you want to convert a specific feature toggle from user/RO specific to global you first have to disable it, then re-enable it as globally

**Comment by shanear** [comment]
Looks awesome
